assets/reader-activation/auth.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Internal dependencies.
+ */
+import './auth.scss';
+
+( function ( readerActivation ) {
+	window.onload = function () {
+		if ( ! readerActivation ) {
+			return;
+		}
+
+		const container = document.querySelector( '#newspack-reader-auth' );
+		if ( ! container ) {
+			return;
+		}
+
+		const initialForm = container.querySelector( 'form' );
+		let form;
+		/** Workaround AMP's enforced XHR strategy. */
+		if ( initialForm.getAttribute( 'action-xhr' ) ) {
+			initialForm.removeAttribute( 'action-xhr' );
+			form = initialForm.cloneNode( true );
+			initialForm.replaceWith( form );
+		} else {
+			form = initialForm;
+		}
+
+		const actionInput = form.querySelector( 'input[name="action"]' );
+		const emailInput = form.querySelector( 'input[name="email"]' );
+		const passwordInput = form.querySelector( 'input[name="password"]' );
+		const submitButtons = form.querySelectorAll( '[type="submit"]' );
+
+		container.querySelector( 'button[data-close]' ).addEventListener( 'click', function ( ev ) {
+			ev.preventDefault();
+			container.style.display = 'none';
+		} );
+
+		const messageContainer = container.querySelector( '.newspack-reader__auth-form__response' );
+
+		const authLinkMessage = container.querySelector( '[data-has-auth-link]' );
+		authLinkMessage.hidden = true;
+
+		const accountLinks = document.querySelectorAll( '.newspack-reader__account-link' );
+		const triggerLinks = document.querySelectorAll( '[data-newspack-reader-account-link]' );
+
+		/**
+		 * Handle reader changes.
+		 */
+		readerActivation.on( 'reader', ( { detail: { email, authenticated } } ) => {
+			emailInput.value = email || '';
+			if ( accountLinks?.length ) {
+				accountLinks.forEach( link => {
+					try {
+						const labels = JSON.parse( link.getAttribute( 'data-labels' ) );
+						link.querySelector( '.newspack-reader-account-link-label' ).innerHTML = authenticated
```

Comment:
No need for HTML there:

```suggestion
						link.querySelector( '.newspack-reader-account-link-label' ).innerText = authenticated
```

Reply:
Fixed in 32cffd9314257290bab489de064f47a10bda5ccc